### ISSUE TITLE
Inifinite loop in 08_rlfinance_3ac :  Cell [9]

### ISSUE THREAD

**Issue by nicolassalok1** [yhilpisch/rl4f#1]
In : ./code/08_rlfinance_3ac.ipynb

Cell 9 : instruction          %time agent.learn(episodes)     :

agent.learn
     calls dqlagent.py -> class DQLAgent :  test function 
          test calls          replay function 
               replay function calls                 self.model.predict

which causes the program to crash...

Anyone can help ?

Cheers ;)

**Comment by yhilpisch** [comment]
Did you install a Python environment with condo based on the YAML file?

What exactly happens?

There are also PyTorch versions of the notebook now in the
GitHub repository. They should all run e.g. in Google Colab.

**Comment by nicolassalok1** [comment]
Hi, 

First thank you for your answer !

I am trying to run it locally so I conda installed the yaml file yes. Then i tried to figure my way out of the bug but in vain...

There is no way the self.model predicts anything...
it crashes and I don't understand why, I have the right packages and modules, there is nothing fancy going on here but still...

The Cell [9] runs undefinetly after episode 1..

The problem as i mentionned is in the function replay of the dqlagent.py file.

Have you got any idea ?


Cheers, 

Nico

**Comment by yhilpisch** [comment]
I have last week given a workshop on the topic and tested again all
notebooks.

So, I can't tell from a distance what the issue is.

**Comment by nicolassalok1** [comment]
Ok I'll run this one on google colab then, IF I find the issue I'll let you know !

**Comment by jaoujal** [comment]
Hi,

I have the same problem for all the notebooks using tensorflow, but with Pytorch it's working very well.

Thank you Yves Hilpisch for this great book.

Regards

**Comment by nicolassalok1** [comment]
Same here, PyTorch is working just fine !

Le dim. 4 mai 2025 à 17:34, Yves Hilpisch ***@***.***> a
écrit :
